keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Monikered.kt

```diff
@@ -19,12 +19,16 @@ package com.netflix.spinnaker.keel.api
 
 import com.fasterxml.jackson.annotation.JsonIgnore
 import com.netflix.spinnaker.keel.model.Moniker
+import com.netflix.spinnaker.keel.model.parseMoniker
 
 /**
  * An object which has a moniker
  */
 interface Monikered : ResourceSpec {
+  val name: String
+
   val moniker: Moniker
+    @JsonIgnore get() = parseMoniker(name)
```

Comment:
The `Monikered` interface now requires a name it can parse into a `Moniker`.

Reply:
Nice.